guides/application_state.md

```diff
@@ -1,6 +1,44 @@
 # Dealing with Application State
 
-As part of Project Firefly, you will have out-of-the-box access to file storage and to a key-value store. 
+Application state could be a pre-defined variable of your action that is accessible across all invocations, or dynamic values or files uploaded by the web users when the app is running. Project Firefly provides the appropriate tools to handle each of these requirements.
+
+## Default parameters
+
+Sometimes you want to bind the same parameter values for all invocations or you just set default values of your action. In either case, you have two different options: setting params at the action level, or at the package level (so all actions in that package can inherit them). These params are set in the `manifest.yaml` file, as the following example.
+
+```yaml
+hello-world:
+  function: actions/hello/index.js
+  runtime: 'nodejs:12'
+  inputs:
+    name: Joe
+```
+
+In many cases, these variables are different depending on the build environment of the app, such as different tenant names in dev, stage, prod, etc. To make it work seamlessly with Git commits, you could store the real value of the variables in the `.env` file (which is not committed to Git), and reference them in the `manifest.yaml` file.
+
+```bash
+# in .env
+NAME=Joe
+```
+
+```yaml
+# in manifest.yaml
+hello-world:
+  function: actions/hello/index.js
+  runtime: 'nodejs:12'
+  inputs:
+    name: $NAME
+```
+
+### Considerations about security
+
+Many developers use the default params as a mechanism to provision actions with the secrets/passwords needed to authenticate against some other systems (databases, services, APIs).
```

Comment:
This paragraph could be misleading in the sense that the mechanism could be perceived as a workaround to manage application secrets at action level, while it is in fact the out-of-the-box solution for secret management within Runtime (and hence the recommended approach).

Reply:
good point, I have updated the text to differentiate Adobe auth vs. third party auth.
Is it clearer?